test/machines/machines_test.go

```diff
@@ -228,81 +183,120 @@ var _ = framework.SigKubeDescribe("Machines", func() {
-			err = wait.Poll(pollInterval, poolTimeout, func() (bool, error) {
-				_, err := f.CAPIClient.ClusterV1alpha1().Clusters(cluster.Namespace).Get(cluster.Name, metav1.GetOptions{})
-				if err != nil {
-					return false, nil
-				}
-				return true, nil
-			})
+			acw := &awsClientWrapper{client: awsClient}
+			f.CreateMachineAndWait(masterMachine, &awsClientWrapper{client: awsClient})
+			machinesToDelete.AddMachine(masterMachine, f, acw)
+
+			By("Collecting master kubeconfig")
+			restConfig, err := f.GetMasterMachineRestConfig(masterMachine, acw)
 			Expect(err).NotTo(HaveOccurred())
 
-			testMachine = utils.TestingMachine(awsCredSecret.Name, cluster.Name, cluster.Namespace)
-			By(fmt.Sprintf("Creating %q machine", testMachine.Name))
-			_, err = f.CAPIClient.ClusterV1alpha1().Machines(testMachine.Namespace).Create(testMachine)
+			// Load actuator docker image to the master node
+			dnsName, err := acw.GetPublicDNSName(masterMachine)
 			Expect(err).NotTo(HaveOccurred())
 
-			// Verify cluster and machine have been deployed
-			err = wait.Poll(pollInterval, timeoutPoolMachineRunningInterval, func() (bool, error) {
-				if _, err := f.CAPIClient.ClusterV1alpha1().Machines(testMachine.Namespace).Get(testMachine.Name, metav1.GetOptions{}); err != nil {
-					log.Info("Waiting for cluster and machine to be created")
-					return false, nil
-				}
-				return true, nil
-			})
+			err = f.UploadDockerImageToInstance(f.ActuatorImage, dnsName)
 			Expect(err).NotTo(HaveOccurred())
 
-			By("Verify AWS instance is running")
-			awsClient, err := awsclient.NewClient(f.KubeClient, awsCredSecret.Name, awsCredSecret.Namespace, region)
+			// Deploy the cluster API stack inside the master machine
+			clusterFramework := framework.NewFrameworkFromConfig(restConfig)
+			By(fmt.Sprintf("Creating %q namespace", testNamespace.Name))
+			_, err = clusterFramework.KubeClient.CoreV1().Namespaces().Create(testNamespace)
+			Expect(err).NotTo(HaveOccurred())
+
+			clusterFramework.DeployClusterAPIStack(testNamespace.Name)
+
+			By("Deploy worker nodes through machineset")
+			masterPrivateIP, err := acw.GetPrivateIP(masterMachine)
 			Expect(err).NotTo(HaveOccurred())
 
-			err = wait.Poll(pollInterval, poolTimeout, func() (bool, error) {
-				log.Info("Waiting for aws instances to come up")
-				runningInstances, err := machineutils.GetRunningInstances(testMachine, awsClient)
+			// Reuse the namespace, secret and the cluster objects
+			clusterFramework.CreateClusterAndWait(cluster)
+			createSecretAndWait(clusterFramework, awsCredSecret)
+
+			workerUserDataSecret, err := utils.WorkerMachineUserDataSecret("workeruserdatasecret", testNamespace.Name, masterPrivateIP)
+			Expect(err).NotTo(HaveOccurred())
+			createSecretAndWait(clusterFramework, workerUserDataSecret)
+
+			workerMachineSet := utils.WorkerMachineSet(awsCredSecret.Name, workerUserDataSecret.Name, cluster.Name, cluster.Namespace)
+			fmt.Printf("workerMachineSet: %#v\n", workerMachineSet)
```

Comment:
Hmmm, should be removed. In the next unless there are blocking reason to update the PR.